ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -629,8 +685,12 @@ private void verifyDataNodeAndDiskEquivalencyInDc(Datacenter dc, String clusterN
       ensureOrThrow(sealedReplicasInClusterMap.equals(sealedReplicasInHelix),
           "Sealed replicas info mismatch for " + "instance " + instanceName);
     }
-    ensureOrThrow(allInstancesInHelix.isEmpty(),
-        "Following instances in Helix not found in the clustermap " + allInstancesInHelix);
+    if (!expectMoreInHelixDuringValidate) {
```

Comment:
can we track the ones that are expected to be extra? I think that is more fine grained than having one variable which may suppress actual stuff that should not be missing.

Reply:
Same in the if-elses below